src/js/tabs/cli.js

```diff
@@ -127,6 +127,25 @@ TABS.cli.initialize = function (callback) {
                 var out_string = textarea.val();
                 self.history.add(out_string.trim());
 
+                if (semver.gte(CONFIG.apiVersion, "1.41.0")) {  // Added in BF4.0
+                    // check and see if we have multiple commands to treat this as a batch
+                    // if so, prepend a "batch start". If the commands don't contain a "save"
+                    // then also append a "batch end".
+                    var checkArray = out_string.split("\n");
+                    if (checkArray.length > 1) {
+                        var hasSave = false;
+                        for (var i = 0; i < checkArray.length; i++) {
+                            if (checkArray[i].trim().toLowerCase() == "save") {
+                                hasSave = true;
+                            }
+                        }
+                        out_string = "batch start\n" + out_string;
```

Comment:
How will this work for users who do not use configurator to handle CLI?

To me, adding hidden magic to the handling of CLI, and forcing users to only use configurator if they want to use the CLI is wrong. If we want to go this route, we should remove CLI from te firmware, and implement it entirely in configurator.

Reply:
This doesn't attempt to solve the problem for the tiny minority of users using CLI outside the configurator.  It also doesn't introduce any changes or breakages for them.  The CLI will behave like it did before this change.  Likewise any custom scripting outside the configurator will be unaffected.